src/bridge-canister/src/operation_store.rs

```diff
@@ -74,6 +75,7 @@ where
     incomplete_operations: CachedStableBTreeMap<OperationId, OperationLog<P>, M>,
     operations_log: StableBTreeMap<OperationId, OperationLog<P>, M>,
     address_operation_map: StableBTreeMap<H160, OperationIdList, M>,
+    memo_operation_map: StableBTreeMap<Memo, OperationId, M>,
```

Comment:
How do you ensure that the memo is unique?

Reply:
As far as I understand, we don't consider this as a problem. Memos are like a user-side addressing for operations. And a user is able to re-use memo like:

was:
`my_memo -> old_operation`

// starting a new operation using the `my_memo`...

become:
`None -> old_operation`
`my_memo -> new_operation`

Reply:
But you never know what the `memo_operation_map` returns in this case. What is the impact of having two different users using the same memo?

Reply:
This is a good point, @ufoscout . If memo map would be user specific it wouldn't be a problem. But since memo's are shared between users it can have unexpected consequences. We could store all operations with the given memo instead of the last one to fix this, it shouldn't be too hard to implement.